apps/src/templates/manageStudents/Table/UsStateColumn/Cell.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import {connect} from 'react-redux';
+
+import {STATE_CODES} from '@cdo/apps/geographyConstants';
+import {EVENTS, PLATFORMS} from '@cdo/apps/metrics/AnalyticsConstants';
+import analyticsReporter from '@cdo/apps/metrics/AnalyticsReporter';
+import {editStudent} from '@cdo/apps/templates/manageStudents/manageStudentsRedux';
+import {selectedSection} from '@cdo/apps/templates/teacherDashboard/teacherSectionsReduxSelectors';
+import {RootState} from '@cdo/apps/types/redux';
+
+import {CellProps} from './interface';
+
+const Cell: React.FC<CellProps> = ({
+  studentId,
+  value,
+  editedValue = '',
+  isEditing = false,
+  // Provided by redux
+  currentUser,
+  section,
+  editStudent,
+}) => {
+  const handleChange = (event: {target: {value: string}}) => {
+    const selectedUsState = event.target.value || null;
+
+    editStudent(studentId, {usState: selectedUsState});
+
+    analyticsReporter.sendEvent(
```

Comment:
Would it be difficult to move the `SECTION_STUDENTS_TABLE_US_STATE_SET` and `SECTION_STUDENTS_TABLE_US_STATE_BULK_SET` to when the save button is clicked? I noticed these events are published whenever a value is selected in the dropdown, so it's easy to fire many events even if the state is ultimately only set after the user clicks save.

Reply:
The real issue is understanding whether the student's U.S. state was selected via the bulk modal or the individual selector. Therefore, the "select U.S. state" events and the following "save" click event together could provide this information. @alivira can give you more details about the idea